ds_test.go

```diff
@@ -147,6 +147,19 @@ func TestQueryRespectsProcess(t *testing.T) {
 	addTestCases(t, d, testcases)
 }
 
+func TestCloseRace(t *testing.T) {
```

Comment:
Has you been able to get the test to reproduce the error? Also, I notice the badger equivalent has a `closed` boolean (https://github.com/ipfs/go-ds-badger/blob/master/datastore.go#L25).

Could you put in some operation after close (e.g. Put after Close) tests and see if it's safe (e.g. no panics)? If it's not safe then I'd put in a similar closed check.

Reply:
Internally, leveldb does keep track of if the database is in a closing or closed state - it just appears that when running `gets` on a versioned checkpoint of the database that begins before closing and spans into closing, you can end up with a race.